Add prettier and format files

lorenzolewis opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### What kind of changes does this PR include?

<!-- Delete any that don’t apply -->

- Changes to Starlight code


#### Description

- Add `prettier` and `prettier-plugin-astro`
- Add prettier config (based on the main Astro repo)
- Format all files in repo

<!--
Here’s what will happen next:
One or more of our maintainers will take a look and may ask you to make changes.
We try to be responsive, but don’t worry if this takes a day or two.
-->

HiDeoo commented:
The plugin autoloading feature from Prettier was [removed](https://github.com/prettier/prettier/pull/14759) in v3 and the Prettier configuration file does not explicitly specify the Prettier plugin for Astro so I don't think it's running on Astro files (which may be why the diff does not contain any Astro file changes).

Altho, I do not think specifying the plugin in the config would be enough as there is currently [a bug](https://github.com/withastro/prettier-plugin-astro/issues/334#issuecomment-1638672927) in Prettier where you have to manually specify the files handled by a plugin for it to format these files, e.g. `prettier --write **/*.astro`.

I wonder if it would not be wise to wait until this issue is fixed before introducing Prettier to the codebase?

lorenzolewis commented:
Thanks @HiDeoo , didn't even think to double-check the Astro files.

I updated the prettier config to include the plugin and added a `format` script that will at least catch it all (from what I can tell). Also got all those `*.astro` files formatted.

Completely understand if you want to hold off on this until https://github.com/prettier/prettier/issues/15079 / https://github.com/withastro/prettier-plugin-astro/issues/334#issuecomment-1638672927 are resolved.

delucis commented on pnpm-lock.yaml:
I’m fairly sure we shouldn’t be formatting the lock file, so should probably ignore that from formatting?

delucis commented on docs/src/content/docs/es/guides/components.mdx:
```diff
@@ -22,9 +22,7 @@ import OtroComponente from '../../componentes/OtroComponente.astro';
-<OtroComponente>
-  Los componentes también pueden contener **contenido anidado**.
-</OtroComponente>
+<OtroComponente>Los componentes también pueden contener **contenido anidado**.</OtroComponente>
```
Ahhh, the joys of auto-formatting content. We often use a shorter line length in code sample code than in regular code so it stays legible on narrower screens. Is it possible to use a shorter line length in `.md` and `.mdx`?

Example of this change overflowing on the left and current version on the right:
<img width="1179" alt="Screenshot comparison" src="https://github.com/withastro/starlight/assets/357379/7bafb043-4779-4b48-aa0b-b4fe372b73dd">

delucis commented on docs/src/content/docs/es/guides/i18n.mdx:
Looking at the code samples now they’re using tabs, we should probably set [`tab-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/tab-size) for Starlight code blocks.

We can do it in a follow-up PR though so we’re not mixing formatting with meaningful code changes.

delucis commented on docs/src/content/docs/fr/reference/configuration.md:
```diff
@@ -155,11 +155,7 @@ sidebar: [
-  | { items: SidebarItem[] }
-  | { autogenerate: { directory: string } }
-);
+} & ({ link: string } | { items: SidebarItem[] } | { autogenerate: { directory: string } });
```
Another example where a shorter line length in content would be better.

delucis reviewed (commented):
Thanks for this @lorenzolewis! Left a few observations. And… of course this is now conflicting 😅 Hopefully easy enough to resolve by reverting the formatting, updating branch, and then re-formatting?

lorenzolewis commented:
I found another magical button on GitHub that apparently closed this one 😅 I'll re-open another one with the suggestions included.
